### PR TITLE
feat(core-flows,medusa,utils,types): adds delivered_quantity to order

### PR DESCRIPTION
**Pull request by riqwan**
what:

- adds delivered_quantity to order

https://github.com/user-attachments/assets/709b1727-08ed-4a88-ae29-38f13540e301

### CODE REVIEW

**Review by carlos-r-l-rodrigues — APPROVED** [review]
🔥

File: packages/core/types/src/order/service.ts [hunk 1]
```diff
@@ -4591,6 +4592,29 @@ export interface IOrderModuleService extends IModuleService {
...
 
+  /**
+   * This method registers a delivery for an order's fulfillment
+   *
+   * @param {RegisterOrderDeliveryDTO} data - The ordes's delivery data.
+   * @param {Context} sharedContext - A context used to share resources, such as transaction manager, between the application and the module.
+   * @returns {Promise<void>} Resolves when the shipment's data are registered successfully.
+   *
+   * @example
+   * await orderModuleService.registerShipment({
```
Comment (olivermrbl) [inline packages/core/types/src/order/service.ts]: nit:
```suggestion
   * await orderModuleService.registerDelivery({
```

**Review by olivermrbl — APPROVED** [review]
Nice work!

File: packages/core/types/src/order/service.ts [hunk 2]
```diff
@@ -4591,6 +4592,29 @@ export interface IOrderModuleService extends IModuleService {
     sharedContext?: Context
   ): Promise<void>
 
+  /**
+   * This method registers a delivery for an order's fulfillment
+   *
+   * @param {RegisterOrderDeliveryDTO} data - The ordes's delivery data.
+   * @param {Context} sharedContext - A context used to share resources, such as transaction manager, between the application and the module.
+   * @returns {Promise<void>} Resolves when the shipment's data are registered successfully.
```
Comment (riqwan) [inline packages/core/types/src/order/service.ts]:
```suggestion
   * @returns {Promise<void>} Resolves when the delivery is registered successfully.
```

File: packages/admin/dashboard/src/routes/orders/order-detail/components/order-activity-section/order-timeline.tsx [hunk 3]
```diff
@@ -222,6 +222,16 @@ const useActivityItems = (order: AdminOrder): Activity[] => {
         children: <FulfillmentCreatedBody fulfillment={fulfillment} />,
       })
 
+      if (fulfillment.delivered_at) {
+        items.push({
+          title: t("orders.activity.events.fulfillment.delivered"),
+          timestamp: fulfillment.delivered_at,
+          children: (
+            <FulfillmentCreatedBody fulfillment={fulfillment} isShipment />
```
Comment (riqwan) [inline packages/admin/dashboard/src/routes/orders/order-detail/components/order-activity-section/order-timeline.tsx]:
```suggestion
            <FulfillmentCreatedBody fulfillment={fulfillment} />
```

**Pull request merged**